jobserver/api.py

```diff
@@ -137,13 +138,50 @@ def post(self, request, *args, **kwargs):
                 # related Jobs manager (ie job_request.jobs)
                 job_data.pop("job_request_id")
 
-                job, created = job_request.jobs.update_or_create(
+                job, created = job_request.jobs.get_or_create(
                     identifier=job_data["identifier"],
                     defaults={**job_data},
                 )
-
                 log.info("Created or updated Job", job=job.id, created=created)
 
+                if created:
+                    # this is the first time job-server has heard about this
+                    # Job so we can move on to further Jobs.  We're knowingly
+                    # skipping potential notifications here to avoid creating
+                    # false positives.
```

Comment:
The case this will miss is where there was a problem handling the job request and the job-runner immediately creates a failed job as a way of passing the error back. We can probably live with that as an edge case for now but worth handling at some point I think.

Reply:
I'd like to surface that to the UI in a different way… I've just not decided how yet!  More pondering required…